openapi.yaml

```diff
@@ -269,6 +269,85 @@ paths:
               schema:
                 $ref: '#/components/schemas/Problem'
 
+  /share:
+    post:
+      summary: Save the given AsyncAPI document.
+      operationId: saveDoc
+      tags:
+        - share
+      requestBody:
+        description: Parse your AsyncAPI document.
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ShareDocumentRequest"
+      responses:
+        "200":
+          description: AsyncAPI document successfully saved.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ShareDocumentResponse"
```

Comment:
In this case you should return only id of saved doc, in the `share/{id}` you should return document as content (string), so we need two JSON Schemas to describe proper responses, one for `post`, second one for `get` methods.